extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
+| storage:object_path           | string    | The object_path for the asset, used along with bucket |
+| storage:requester_pays        | bool      | Is the data requester pays or is it data manager/cloud provider pays |
+| storage:tier                  | string    | The title for the tier type (as defined by PaaS provider) |
+| storage:tier_duration         | integer   | Minimum storage duration required before additional fees |
```

Comment:
tier_duration seems like we're putting the cost structure schedule of the cloud provider into the metadata of the stac item. I'm not sure what the usefulness of this is.

Reply:
all cloud providers have a concept of a duration that a storage object is required to remain at that storage level before it can be retrieved without cost. It's not as precise as cost structure information, but it's similar to `requester_pays` in importance.

Reply:
I get requester_pays, since that has to be set in the request or you can't retrieve the item. I don't know if that's ever used to make a decision about whether it should be retrieved or not (e.g., don't retrieve anything you have to pay for).  How would tier_duration be used? I just want to know what the use case is.